osquery/tables/system/darwin/asl.cpp

```diff
@@ -0,0 +1,32 @@
+#include <asl.h>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "asl_utils.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genAsl(QueryContext &context) {
+  QueryData results;
+
+  asl_object_t query = createAslQuery(context);
+  asl_object_t result = asl_search(NULL, query);
```

Comment:
Let's use `nullptr` for all the null pointers, it seems like the first param is: `struct __asl_object_s *`.